keras/src/quantizers/gptq_core.py

```diff
@@ -0,0 +1,335 @@
+import random
+
+import numpy as np
+from absl import logging
+
+from keras.src import ops
+from keras.src import utils as keras_utils
+from keras.src.layers import Dense
+from keras.src.layers import EinsumDense
+from keras.src.layers import Embedding
+from keras.src.quantizers.gptq import GPTQ
+from keras.src.quantizers.gptq_quant import GPTQQuantization
+
+
+def get_dataloader(tokenizer, sequence_length, dataset, num_samples=128):
+    """
+    Prepares and chunks the calibration dataloader, repeating short datasets.
+    """
+    all_tokens = []
+
+    if isinstance(dataset, str):
+        raise TypeError(
+            "The `dataset` argument must be an iterable (e.g., a list of "
+            "strings or a generator). Providing a dataset name as a string "
+            "is not supported. Please pass the loaded dataset directly."
+        )
+
+    logging.info("Using pre-made dataset/generator...")
```

Comment:
`logging.info("Using pre-made dataset/generator...")`

This is no longer relevant I believe, you can remove

Reply:
Done